File: Frontend/moveinn/src/types/recommendation.ts
import { ReactNode } from "react";

export interface Recommendation {
  id: string;
  title: string;
  description: string;
  address: string;
  city: string;
  country: string;
  createdAt: string;
  rating: number;
  category: number;
  tags: string[];
  userId: string;
  recommendationImages: {
    id: string;
    url: string;
  }[];
}
export interface DetailedRecommendationCardProps {
  recommendation: Recommendation;
  categoryIcon: ReactNode;
}
export const categories = {
  Restaurant: 0,
  Cafeteria: 1,
  Museum: 2,
  LeisureZone: 3,
  Park: 4,
  HistoricalSite: 5,
  Shopping: 6,
  Bar: 7,
  Other: 8,
} as const;

File: Backend/python-backend/schemas/recommendation.py
from pydantic import BaseModel
from typing import Optional, List
from enums import Category, Rating

class RecommendationBase(BaseModel):
    title: str
    description: Optional[str] = None
    category: Optional[Category] = None
    address: str
    city: str
    country: str
    rating: Optional[Rating] = None
    created_at: Optional[str] = None  # ISO date string
    tags: Optional[List[str]] = None
    user_id: Optional[str] = None

class RecommendationCreate(RecommendationBase):
    pass

class RecommendationRead(RecommendationBase):
    id: str
    recommendation_images: Optional[List[str]] = None  # List of image URLs or IDs

class RecommendationUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    category: Optional[Category] = None
    address: Optional[str] = None
    city: Optional[str] = None
    country: Optional[str] = None
    rating: Optional[Rating] = None
    created_at: Optional[str] = None
    tags: Optional[List[str]] = None
    user_id: Optional[str] = None
    recommendation_images: Optional[List[str]] = None 